packages/@react-stately/color/src/Color.ts

```diff
@@ -77,13 +76,33 @@ export class Color {
 
     throw new Error('Unsupported color channel: ' + channel);
   }
+
+  getChannelValuePercent(channel: ColorChannel): number {
+    return this.value.getChannelValuePercent(channel);
+  }
+
+  withChannelValue(channel: ColorChannel, value: number): Color {
+    let x = Color.fromColorValue(this.value.clone());
+    x.value.setChannelValue(channel, value);
+    return x;
+  }
+
+  withChannelValuePercent(channel: ColorChannel, value: number): Color {
+    let x = Color.fromColorValue(this.value.clone());
+    x.value.setChannelValuePercent(channel, value);
+    return x;
+  }
 }
 
 interface ColorValue {
   toRGB(): ColorValue,
   toHSB(): ColorValue,
   toHSL(): ColorValue,
-  toString(format: ColorFormat): string
+  toString(format: ColorFormat | 'css'): string,
+  clone(): ColorValue,
+  setChannelValue(format: ColorChannel, value: number): void,
+  setChannelValuePercent(format: ColorChannel, value: number): void,
+  getChannelValuePercent(format: ColorChannel): number
```

Comment:
Rather than implementing the percent-based versions on every color value implementation separately, what do you think about implementing `getChannelMin` and `getChannelMax` as proposed in #973 and doing the normalization in the `Color` class instead? We will need these min/max values for other purposes (e.g. properties of a slider element), so maybe it makes sense to centralize around that?

Reply:
What about the normalization of the degree value? Move that to the colorwheel stately hook?